modules/kairos-k3s-cluster/server.tf

```
@@ -66,13 +66,13 @@ module "k3s_server_vm" {
     })
     bundles = file("${path.module}/files/bundles.yaml")
     write_files = templatefile("${path.module}/templates/user-data/write-files.yaml.tftpl", {
-      files = [{
-        path        = "/var/lib/rancher/k3s/server/manifests/${var.vault_auth_service_account}.yaml"
+      files = concat(var.additional_manifests, [{
+        dest        = "/var/lib/rancher/k3s/server/manifests/${var.vault_auth_service_account}.yaml"
         permissions = 0644
         content = templatefile("${path.module}/templates/user-data/write-files/vault-auth.yaml.tftpl", {
           vault_auth_sa = var.vault_auth_service_account
         })
-      }]
+      }])
```

```suggestion
      files = concat(local.formatted_additional_manifests, [{
        dest        = "${local.k3s_manifest_dir}/${var.vault_auth_service_account}.yaml"
        permissions = 0644
        content = templatefile("${path.module}/templates/user-data/write-files/vault-auth.yaml.tftpl", {
          vault_auth_sa = var.vault_auth_service_account
        })
      }])
```

can't make suggestions in `locals.tf` as not edited in this PR, but here is what I have in locals
```
  manifest_default_permissions = 0644
  k3s_manifest_dir             = "/var/lib/rancher/k3s/server/manifests/"

  formatted_additional_manifests = [
    for m in var.additional_manifests : {
      dest        = "${local.k3s_manifest_dir}/m.name"
      permissions = m.permissions != null ? m.permissions : local.manifest_default_permissions
      content     = m.content
    }
  ]
```

---

(I've pushed all these suggestions to `AndEsterson/additional-manifests` if easier to look at that way

---

Good idea with the locals - thanks!